src/containers/tud/TimeUseDiaryContainer.js

```diff
@@ -85,6 +109,36 @@ const TimeUseDiaryContainer = () => {
     }
   }, [page, formSchema]);
 
+  const refreshProgress = (currFormData) => {
+    const dayStart = selectTimeByPageAndKey(DAY_SPAN_PAGE, DAY_START_TIME, currFormData);
+    const dayEnd = selectTimeByPageAndKey(DAY_SPAN_PAGE, DAY_END_TIME, currFormData);
+    const currentEnd = selectTimeByPageAndKey(page, ACTIVITY_END_TIME, currFormData);
+
+    setDayStartTime(dayStart);
+    setDayEndTime(dayEnd);
+    setCurrentTime(currentEnd);
+  };
+
+  /* eslint-disable react-hooks/exhaustive-deps */
+  useEffect(() => {
+    refreshProgress(formData);
+  }, [page]);
+
+  useEffect(() => {
+    setIsSummaryPage(getIsSummaryPage(formData, page));
+  }, [page]);
```

Comment:
hmm could these tasks be done within the same `useEffect` hook, since they each have the same dependency?

Reply:
um, yes. It certainly makes sense to have them in one hook.